src/stdlib_sorting_sort.fypp

```diff
@@ -0,0 +1,401 @@
+
+            do j=1_int_size, size(array, kind=int_size)-1
+                key = array(j)
+                i = j - 1
+                do while( i >= 0 )
+                    if ( array(i) <= key ) exit
+                    array(i+1) = array(i)
+                    i = i - 1
+                end do
+                array(i+1) = key
+            end do
+
+        end subroutine insertion_sort
+
+        pure subroutine heap_sort( array )
+! A bog standard heap sort
+            ${t1}$, intent(inout) :: array(0:)
+
+            integer(int_size) :: i, heap_size
+            ${t1}$   :: y
+
+            heap_size = size( array, kind=int_size )
+! Build the max heap
+            do i = (heap_size-2)/2_int_size, 0_int_size, -1_int_size
+                call max_heapify( array, i, heap_size )
+            end do
+            do i = heap_size-1, 1_int_size, -1_int_size
+! Swap the first element with the current final element
+                y = array(0)
+                array(0) = array(i)
+                array(i) = y
+! Sift down using max_heapify
+                call max_heapify( array, 0_int_size, i )
+            end do
+
+        end subroutine heap_sort
+
+        pure recursive subroutine max_heapify( array, i, heap_size )
+! Transform the array into a max heap
+            ${t1}$, intent(inout) :: array(0:)
+            integer(int_size), intent(in)  :: i, heap_size
+
+            integer(int_size) :: l, r, largest
+            ${t1}$   :: y
+
+            largest = i
+            l = 2_int_size * i + 1_int_size
+            r = l + 1_int_size
+            if ( l < heap_size ) then
+                if ( array(l) > array(largest) ) largest = l
+            end if
+            if ( r < heap_size ) then
+                if ( array(r) > array(largest) ) largest = r
+            end if
+            if ( largest /= i ) then
+                y = array(i)
+                array(i) = array(largest)
+                array(largest) = y
+                call max_heapify( array, largest, heap_size )
+            end if
+
+        end subroutine max_heapify
+
+    end subroutine ${k1}$_sort
+
+#:endfor
+
+
+
+
+    pure module subroutine char_sort( array )
+! `char_sort( array )` sorts the input `ARRAY` of type `CHARACTER(*)`
+! using a hybrid sort based on the `introsort` of David Musser. As with
+! `introsort`, `char_sort( array )` is an unstable hybrid comparison
+! algorithm using `quicksort` for the main body of the sort tree,
+! supplemented by `insertion sort` for the outer brances, but if
+! `quicksort` is converging too slowly the algorithm resorts
+! to `heapsort`. The algorithm is of order O(N Ln(N)) for all inputs.
+! Because it relies on `quicksort`, the coefficient of the O(N Ln(N))
+! behavior is typically small compared to other sorting algorithms.
+
+        character(len=*), intent(inout) :: array(0:)
+
+        integer(int32) :: depth_limit
+
+        depth_limit = 2 * int( floor( log( real( size( array, kind=int64 ),  &
```

Comment:
```suggestion
        depth_limit = 2 * int( floor( log( real( size( array, kind=int_size ),  &
```

Reply:
I want to fetch and merge (pull) the most recent changes into my local repository. what should I use for the <repository> in the `git pull` command?

Reply:
in your local directory, assuming that `origin` refers to `https://github.com/wclodius2/stdlib`, run:
```shell
git pull origin sorting2
```
to get the latest changes done in the remote repo.

Reply:
Note that you can first commit my suggestions, directly by blicking on the button "commit suggestion".